apps/accounts/automations/change_secret/manager.py

```diff
@@ -172,7 +176,7 @@ def on_host_success(self, host, result):
             print("Account not found, deleted ?")
             return
 
-        version_update_required = account.secret != recorder.new_secret
+        version_update_required = self.require_update_version(account, recorder)
         account.secret = recorder.new_secret
         account.date_updated = timezone.now()
 
```

Comment:
There aren't any significant code differences between the snippets provided. Both of them seem to be functioning correctly as designed in terms of function calls and logic flow.
Optimization Suggestions: No immediate optimizations can be made without more context about specific performance metrics or inefficiencies.